core/dbt/task/run.py

```diff
@@ -436,8 +436,9 @@ def defer_to_manifest(self, adapter, selected_uids: AbstractSet[str]):
 
     def before_run(self, adapter, selected_uids: AbstractSet[str]):
         with adapter.connection_named("master"):
-            self.create_schemas(adapter, selected_uids)
-            self.populate_adapter_cache(adapter)
+            required_schemas = self.get_model_schemas(adapter, selected_uids)
```

Comment:
@jtcohen6 `get_model_schemas` would remove all of the node that is not in `selected_uids`, is there a case we need adapter cache to contain more nodes than `selected_nodes`?

Reply:
@ChenyuLInx I'm not sure! That's _the_ critical question animating in this issue/PR. I will say, having this PR makes it a lot easier to test.

- **Will dbt return _incorrect_ results, if it fails to cache an unselected relation?** I don't think so, as long as caching happens consistently at the schema level. (That's an open question for us right now on Spark: https://github.com/dbt-labs/dbt-spark/issues/296)
- **How will this work for defer?** When `--defer` is enabled, it's necessary to call `adapter.get_relation` for _all_ unselected parents of selected models, to figure out whether to use a "dev" version or defer to the "prod" version. In my local testing, this still works! dbt will just record a cache miss, and run the query independently:
```
08:35:23.500589 [debug] [MainThread]: On "master": cache miss for schema "jerco.dbt_jcohen_schema_a", this is inefficient
```
It's likely that this will be less efficient. So, I think it's important that the behavior remains configurable. I'm not opposed to moving forward with this, in order to test the waters. We may someday wish to turn it on by default.